packages/qwik-city/runtime/src/qwik-city-component.tsx

```diff
@@ -514,6 +515,9 @@ export const QwikCityProvider = component$<QwikCityProps>((props) => {
             const scrollState = currentScrollState(scroller);
             saveScrollHistory(scrollState);
             win._qCityScrollEnabled = true;
+            if (isBrowser) {
+              callRestoreScrollOnDocument();
+            }
```

Comment:
In V1 this logic is inside the qwik core in the notify-render. I don't like this, so I moved this here (tests are passing, so I guess it is ok)

Reply:
The only thing is transition api, there is no test for it